### PR TITLE
GHCR image push-restic-backup-restore

### PR DESCRIPTION
**Pull request by Praveenraj-K**
-Pushing the images in both Docker Hub & GitHub Container Registry "GHCR.io"
-Only initiate when there is a commit in the main branch using the workflow trigger.

### CODE REVIEW

**Comment by devon-sil** [comment]
@Praveenraj-K Thanks for putting together a PR for this. I've added the developers for review.

File: .github/workflows/build-and-publish.yml [hunk 1]
```diff
@@ -2,10 +2,12 @@ name: Build and Publish
 
 on:
   push:
+    branches:
+      - main
```
Comment (briskt) [inline .github/workflows/build-and-publish.yml]: This should include `tags` as well

File: .github/workflows/build-and-publish.yml [hunk 2]
```diff
@@ -17,13 +19,25 @@ jobs:
...
       - name: Extract metadata (tags, labels) for Docker
         id: meta
         uses: docker/metadata-action@v5
         with:
-          images: ${{ vars.DOCKER_ORG }}/${{ github.event.repository.name }}
+          images: |   
+            ${{ vars.DOCKER_ORG }}/${{ github.event.repository.name }}
+            ghcr.io/${{ github.repository }}
+          tags: |
+            type=ref,event=branch  # Tag with branch name
```
Comment (briskt) [inline .github/workflows/build-and-publish.yml:37]: This should include `semver` or `type=ref,event=tag` as well.

**Review by briskt — CHANGES_REQUESTED** [review]
same comments as other repos

**Review by briskt — APPROVED** [review]

**Pull request merged**